checker/src/visitors.rs

```diff
@@ -87,6 +87,7 @@ pub struct MirVisitor<'analysis, 'compilation, 'tcx, E> {
     generic_arguments: Option<Vec<ExpressionType>>,
     heap_addresses: HashMap<mir::Location, Rc<AbstractValue>>,
     post_condition: Option<Rc<AbstractValue>>,
+    post_condition_block: Option<mir::BasicBlock>,
```

Comment:
It may be better to put the value and corresponding block into a tuple.

Reply:
I tried this in a first version, and concluded flattening it out into the global type leads to less boilerplate. We can look at it if we have more usage of the pair (Cond, BasicBlock).